File: CoffeeTracker-WebAPI/CoffeeTracker/Models/CoffeeCups.cs
using System.ComponentModel.DataAnnotations;

namespace CoffeeTracker.Models;

public class CoffeeCups
{
    public int Id {get; set;}
    public int Quantity {get; set;}
    public int? Measure {get; set;}
    public string? Description {get; set;}

    [EnumDataType(typeof(CoffeeMeasureUnits))]
    public CoffeeMeasureUnits Units { get; set;}

    [DataType(DataType.Date)]
    public DateTime Date {get; set;}
}

File: CoffeeTracker/src/app/coffee-cups.ts
import { Form, FormControl } from "@angular/forms";

export interface CoffeeCups {
    id: number;
    quantity: number;
    measure: number;
    description: string;
    units: CoffeeMeasureUnits;
    date: Date
}

export interface CoffeeCupsForm {
    id: FormControl<number>;
    quantity: FormControl<number>;
    measure: FormControl<number>;
    description: FormControl<string>;
    units: FormControl<CoffeeMeasureUnits>;
    date: FormControl<Date>;
  }

export enum CoffeeMeasureUnits {
    ml = 0,
    oz = 1
}